src/ztex/fpga-sha256crypt/README.md

```diff
@@ -0,0 +1,143 @@
+- SHA256 instructions store instruction data in internal buffer
+(procb_buf). A dedicated unit (process_bytes) takes intermediate data,
+fetches input data from the memory, creates 16 x 32-bit data blocks
+for cores, adds padding and total where necessary. It saves
+the state of an unfinished computation, switches to the next core
+after each block.
+- The program for the CPU is available <a href='https://github.com/magnumripper/JohnTheRipper/blob/bleeding-jumbo/src/ztex/fpga-sha256crypt/sha256crypt/cpu/program.vh'>here</a>.
+
+
+## Design overview
+
+```
+       ------------------+--------+--+--+--------+---------
+      /                 /        /  /  /        /          \
+ +-----------+   +-----------+             +-----------+    |
+ |           |   |           |   X  X  X   |           |    |
+ | Computing |   | Computing |             | Computing |    |
+ |  Unit #0  |   |  Unit #1  |             |  Unit N   |    |
+ |           |   |           |   X  X  X   |           |    |
+ +-----------+   +-----------+             +-----------+    |
+       ^               ^         ^  ^  ^         ^          |
+       |               |         |  |  |        /           |
+       +---------------+---------+--+--+--------            |
+       |                                                    |
+ . . . | . . . . . . . . . . . . . . . . . . . . . . . . . .| .
+       |                                                    |
+       |                                                    |
+  +-----------------+          +----------------+          /
+  |     Arbiter     |          |     Arbiter    |<---------
+  | (transmit part) |<-------->| (receive part) |
+  +-----------------+          +----------------+
+       ^                                    |
+       |                    +------------+  |  +------
+       |      +---------+   |            |<-+->| mode \
+       |      | cmp.    |-->| comparator |     | cmp   |--
+       |      | config. |   |            |---->| ?    /   \
+       |      +---------+   +------------+     +------     |
+       |          ^                                        |
+ . . . | . . . . .|. . . . . . . . . . . . . . . . . . . . | .
+       |          |      Communication framework           |
+       |          |                                       /
+  +-----------+   |                                      /
+  | candidate |   |                                     /
+  | generator |   |                                    /
+  +-----------+---------+----------------------+      /
+  | input pkt. handling | output pkt. creation |<-----
+  +---------------------+----------------------+
+  | input FIFO          | output FIFO          |
+  +---------------------+----------------------+
+  |  Prog. clocks  | USB I/O   |  FPGA reset   |
+  +--------------------------------------------+
+
+```
+fig.2. Overview, FPGA application
+
+- Each FPGA has 25 computing units, that's 75 cores, 150 keys are
+computed in parallel.
+
```

Comment:
It would be helpful to add (with the separate commit/PR) some info on relative resource usage by SHA-256 cores vs. the "overhead" (CPUs, glue logic). We do have this info in sha512crypt-ztex's README.